app/src/main/java/org/simple/clinic/enterotp/EnterOtpEffectHandler.kt

```diff
@@ -44,6 +45,7 @@ class EnterOtpEffectHandler @AssistedInject constructor(
         .addConsumer(FailedLoginOtpAttempt::class.java, { bruteForceProtection.incrementFailedOtpAttempt() }, schedulers.io())
         .addAction(ShowNetworkError::class.java, uiActions::showNetworkError, schedulers.ui())
         .addAction(ShowUnexpectedError::class.java, uiActions::showUnexpectedError, schedulers.ui())
+        .addAction(AllowOtpEntry::class.java, { uiActions.showOtpEntryMode(OtpEntry) }, schedulers.ui())
```

Comment:
Is there a reason why this is not part of the model and UI renderer? One of the reasons why that is important is, UI actions don't get restored after state restoration or process death.